logic/src/commonTest/kotlin/com/wire/kalium/logic/data/call/CallRepositoryTest.kt

```diff
@@ -66,63 +93,100 @@ class CallRepositoryTest {
+        }
     }
 
     @Test
-    fun givenACallObject_whenCreateCallCalled_thenAddThatCallToTheFlow() {
+    fun givenACallObject_whenCreateCallCalled_thenAddThatCallToTheFlow() = runTest {
+        given(conversationRepository).suspendFunction(conversationRepository::getConversationDetailsById)
+            .whenInvokedWith(any())
+            .thenReturn(flowOf(ConversationDetails.Group(TestConversation.ONE_ON_ONE, LegalHoldStatus.ENABLED)))
+        given(userRepository).suspendFunction(userRepository::getKnownUser)
+            .whenInvokedWith(any())
+            .thenReturn(flowOf(TestUser.OTHER))
+        given(teamRepository).suspendFunction(teamRepository::getTeam)
+            .whenInvokedWith(any())
+            .thenReturn(flowOf(Team("team1", "team_1")))
         callRepository.updateCallProfileFlow(CallProfile(mapOf(startedCall.conversationId.toString() to startedCall)))
 
-        callRepository.createCall(answeredCall)
+        callRepository.createCall(conversationIdAnsweredCall, CallStatus.ANSWERED, "caller_id")
 
         val calls = callRepository.callsFlow()
-        assertEquals(2, calls.value.size)
-        assertEquals(calls.value[0], startedCall)
-        assertEquals(calls.value[1], answeredCall)
+
+        calls.test {
+            val list = awaitItem()
+            assertEquals(2, list.size)
+            assertEquals(list[0], startedCall)
+            assertEquals(list[1], answeredCall)
+        }
     }
 
     @Test
-    fun givenACallObjectWithSameConversationIdAsAnotherOneInTheFlow_whenCreateCallCalled_thenReplaceTheCurrent() {
+    fun givenACallObjectWithSameConversationIdAsAnotherOneInTheFlow_whenCreateCallCalled_thenReplaceTheCurrent() = runTest {
+        given(conversationRepository).suspendFunction(conversationRepository::getConversationDetailsById)
+            .whenInvokedWith(any())
+            .thenReturn(flowOf(ConversationDetails.Group(TestConversation.ONE_ON_ONE, LegalHoldStatus.ENABLED)))
+        given(userRepository).suspendFunction(userRepository::getKnownUser)
+            .whenInvokedWith(any())
+            .thenReturn(flowOf(TestUser.OTHER))
+        given(teamRepository).suspendFunction(teamRepository::getTeam)
+            .whenInvokedWith(any())
+            .thenReturn(flowOf(Team("team1", "team_1")))
+
         val incomingCall2 = provideCall(sharedConversationId, CallStatus.INCOMING)
         callRepository.updateCallProfileFlow(CallProfile(mapOfCallProfiles))
 
-        callRepository.createCall(incomingCall2)
+        callRepository.createCall(sharedConversationId, CallStatus.INCOMING, "caller_id")
 
         val calls = callRepository.callsFlow()
-        assertEquals(mapOfCallProfiles.size, calls.value.size)
-        assertEquals(calls.value[0], incomingCall2)
+        calls.test {
+            val list = awaitItem()
+            assertEquals(mapOfCallProfiles.size, list.size)
+            assertEquals(list[0], incomingCall2)
+        }
```

Comment:
same here...
we should verify that getConversationDetailsById, getKnownUser and getTeam are being invoked once